### PR TITLE
Fix Trainee Page Modal

### PR DESCRIPTION
**Pull request by arthurjunior250**
**What does this PR do?**  
-A user will be able to close the modal  without reloading functionalities
Screenshot_2022-11-16_10-37-25

### CODE REVIEW

File: src/hooks/useDarkMode.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
```
Comment (capitale250) [inline src/hooks/useDarkMode.ts]: why are we having this hook ,we allready have one

**Review by capitale250 — CHANGES_REQUESTED** [review]

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Good!

1. Remove `package-lock.json` from the project
2. Not sure if a toast is necessary each time you refresh a page.

File: src/pages/TraineApplicant/Trainee.tsx [hunk 2]
```diff
@@ -80,7 +90,46 @@ const Open=() =>{
...
+    setFirstname("");
+    setLastname("");
+    setEmail("");
+    //   toast.success("succeed")
+    setAddNewTraineeModel(false);
+    setTimeout(reload, 3000);
+  }
+};
 
+function reload() {
```
Comment (capitale250) [inline src/pages/TraineApplicant/Trainee.tsx]: why are we still having reload function ,that is not what we need

**Review by capitale250 — CHANGES_REQUESTED** [review]

File: src/pages/TraineApplicant/Trainee.tsx [hunk 3]
```diff
@@ -80,6 +89,43 @@ const Open=() =>{
...
+    email: email,
+    cycle_id: cycle_id,
+  };
+  if (props.createTrainee(data)) {
+    setFirstname("");
+    setLastname("");
+    setEmail("");
+    setAddNewTraineeModel(false);
+    setTimeout(() => {
+      window.location.reload();
```
Comment (capitale250) [inline src/pages/TraineApplicant/Trainee.tsx]: we don't need to reload

**Review by capitale250 — CHANGES_REQUESTED** [review]

File: src/pages/TraineApplicant/Trainee.tsx (hunk 3 above)
Comment (UmuhireJessie) [reply src/pages/TraineApplicant/Trainee.tsx]: > we don't need to reload

+1

**Review by UmuhireJessie — COMMENTED** [review]
Conside removing `window.location.reload();`

File: src/pages/TraineApplicant/Trainee.tsx [hunk 4]
```diff
@@ -158,26 +292,28 @@ const AddTrainee = (props: any) => {
...
-                                          <td className="px-5 py-5 border-b border-gray-200 dark:border-dark-tertiary text-sm">
-                                            <div className="flex items-center">
-                                              <div className="">
-                                                <p className="text-gray-900 text-center dark:text-white whitespace-no-wrap">
-                                                  {item.lastName}
-                                                </p>
-                                              </div>
-                                            </div>
-                                          </td>
+                                  item.delete_at == false ? (
```
Comment (capitale250) [inline src/pages/TraineApplicant/Trainee.tsx:293]: we needed that check of item

File: src/redux/actions/deletetraine.ts [hunk 5]
```diff
@@ -150,4 +152,71 @@ export const fetchtraine = ({ page,itemsPerPage,  All }:any) => {
                   dispatch({type: fetchtrainesss.fetchtraines_fail,error:err})})
         } catch(err:any) {toast.error(err.message);dispatch({type: fetchtrainesss.fetchtraines_fail,error:err})}
     }
-}
\ No newline at end of file
+}
+export const createtraine = ({ firstName, lastName, email, cycle_id }: any) => {
+    return async (dispatch: Dispatch<fetchact>) => {
```
Comment (capitale250) [inline src/redux/actions/deletetraine.ts:157]: why have we created other actions?

**Review by capitale250 — CHANGES_REQUESTED** [review]
why have we created new reducers and acrtions in redux

**Review by mugiisha — APPROVED** [review]

**Review by gerard1001 — COMMENTED** [review]
Thanks @arthurjunior250, Everything looks good. 
Though, you can remove the functions and actions you won't be using.

File: src/redux/actions/deletetraine.ts (hunk 5 above)
Comment (arthurjunior250) [reply src/redux/actions/deletetraine.ts:157]: the first one is not being used on trainee page

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**